### PR TITLE
ISSUE-89 All simulation

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
```
[error] /home/travis/build/linagora/james-gatling/src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala:23:34: not found: value recipientFeeder
[error]     .exec(JmapEmail.submitEmails(recipientFeeder))
[error]                                  ^
[warn] /home/travis/build/linagora/james-gatling/src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/SelectMailboxSimulation.scala:5:52: Unused import
[warn] import org.apache.james.gatling.jmap.draft.scenari.JmapSelectArbitraryMailboxScenario
[warn]                                                    ^
[warn] one warning found
[error] one error found
[error] (Test / compileIncremental) Compilation failed
[error] Total time: 10 s, completed Nov 20, 2020, 4:01:16 AM
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
The command "sbt test" exited with 1.
```

Can you fix this @LanKhuat ?

File: src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import scala.concurrent.duration._
+
+class AllSimulation extends Simulation with SimulationOnMailCorpus {
+  private val MIN_MESSAGES_IN_MAILBOXES_TO_SELECT = 100
+
+  def emailSubmissionScenario(userFeeder: UserFeederBuilder, recipientFeeder: RecipientFeederBuilder): ScenarioBuilder = scenario("EmailSubmissionScenario")
+    .feed(userFeeder)
+    .exec(retrieveAccountId)
+    .exec(CommonSteps.provisionSystemMailboxes())
```
Comment (Arsnael) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala:24]: I think we might want to check simulations in general, but we don't need to provision system mailboxes on the provisioning, as we already have them

File: src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.james.gatling.jmap.rfc8621.scenari.{EmailKeywordsUpdatesScenario, InboxLoadingScenario, OpenEmailScenario, SelectMailboxScenario}
+import org.apache.james.gatling.jmap.{InboxHomeLoading, OpenMessage, SelectMailbox}
+import org.apache.james.gatling.simulation.SimulationOnMailCorpus
+
+import scala.concurrent.duration._
+
+class AllSimulation extends Simulation with SimulationOnMailCorpus {
+  private val MIN_MESSAGES_IN_MAILBOXES_TO_SELECT = 100
+
+  def emailSubmissionScenario(userFeeder: UserFeederBuilder, recipientFeeder: RecipientFeederBuilder): ScenarioBuilder = scenario("EmailSubmissionScenario")
```
Comment (Arsnael) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala:21]: Why not reusing the scenario that was developed for that in the previous PR?

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/AllScenario.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import io.gatling.core.structure.ScenarioBuilder
+import io.gatling.http.Predef._
+import org.apache.james.gatling.control.RecipientFeeder.RecipientFeederBuilder
+import org.apache.james.gatling.control.UserFeeder.UserFeederBuilder
+import org.apache.james.gatling.jmap.draft.CommonSteps
+import org.apache.james.gatling.jmap.rfc8621.{JmapEmail, JmapHttp, SessionStep}
+
+import scala.concurrent.duration._
+
+class AllScenario {
```
Comment (Arsnael) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/AllScenario.scala]: this scenario is not being used by any simulation?

File: src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala (hunk 2 above)
Comment (LanKhuat) [reply src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala:21]: we want to submit just once, not with the during part

**Review by chibenwa — APPROVED** [review]

File: src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package org.apache.james.gatling.simulation.jmap.rfc8621
+
+import io.gatling.core.Predef._
+import io.gatling.core.scenario.Simulation
+import io.gatling.core.structure.ScenarioBuilder
+import org.apache.james.gatling.control.RecipientFeeder
+import org.apache.james.gatling.control.RecipientFeeder.RecipientFeederBuilder
+import org.apache.james.gatling.control.UserFeeder.UserFeederBuilder
+import org.apache.james.gatling.jmap.draft.CommonSteps
```
Comment (chibenwa) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/AllSimulation.scala:9]: Please open another PR to provide a rfc-8621 version of this

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**